opm/simulators/flow/GenericOutputBlackoilModule.cpp

```diff
@@ -391,6 +391,54 @@ outputFipAndResvLog(const Inplace& inplace,
     }
 }
 
+template<class FluidSystem>
+void GenericOutputBlackoilModule<FluidSystem>::
+accumulateRftDataParallel(const Parallel::Communication& comm) {
+    if (comm.size() > 1) {
+        collectRftMapOnRoot(oilConnectionPressures_, comm);
+        collectRftMapOnRoot(waterConnectionSaturations_, comm);
+        collectRftMapOnRoot(gasConnectionSaturations_, comm);
+    }
+}
+
+template<class FluidSystem>
+void GenericOutputBlackoilModule<FluidSystem>::
+collectRftMapOnRoot(std::map<std::size_t, Scalar>& local_map, const Parallel::Communication& comm) {
+
+    const auto mapsize = local_map.size();
+    std::vector<int> keys;
+    std::vector<Scalar> values;
+    keys.reserve(mapsize);
+    values.reserve(mapsize);
+    for (const auto& [key, value] : local_map) {
+        keys.push_back(key);
+        values.push_back(value);
+    }
+
+    std::vector<int> all_keys;
+    std::vector<Scalar> all_values;
+    std::vector<int> offsets;
+
+    std::tie(all_keys, offsets) = Opm::gatherv(keys, comm, 0);
+    std::tie(all_values, std::ignore) = Opm::gatherv(values, comm, 0);
+    assert(all_keys.size() == all_values.size());
+
+    // Insert/update map values on root
+    if (comm.rank() == 0) {
+        for (auto i=static_cast<std::size_t>(offsets[1]); i<all_keys.size(); ++i) {
+            const auto index = all_keys[i];
+            if (local_map.count(index)>0) {
+                const Scalar prev_value = local_map[index];
+                local_map[index] = std::max(prev_value, all_values[i]);
```

Comment:
Maybe prevent multiple searches?
```suggestion
            if (auto candidate = local_map.find(index); candidate != local_map.end() && candidate->first == index) {
                const Scalar prev_value = candidate->second;
                candidate->second = std::max(prev_value, all_values[i]);
```
Will max do the correct thing on all occasions?

Reply:
I believe so - the gathered values are set via processElement via EclWriter, which iterates only over the interior, so if my understanding is correct the only non-zero value comes from the partition to which the connection truly belongs?